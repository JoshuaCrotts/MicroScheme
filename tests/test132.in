;
; Data structure k/v pair testing 2
;

(define (caar l)
    (car (car l))
)

(define (assoc key data)
    (cond ((null? data) #f)
          ((equal? (caar data) key) (car data))
          (else (assoc key (cdr data)))
    )
)

(define price-list
 '( 
    ((bmw m5) 71095)
    ((bmw z4) 40495)
    ((jag xj8) 56975)
    ((mb sl500) 86655)
  )
)
(assoc '(bmw z4) price-list)
; ((bmw z4) 40495)
