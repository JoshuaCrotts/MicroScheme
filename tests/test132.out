((bmw z4) 40495)
